src/jit/lsra.cpp

```diff
@@ -5608,7 +5608,22 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
         if (physRegRecord->assignedInterval == nullptr ||
             physRegRecord->assignedInterval->getNextRefLocation() > lastLocation)
         {
-            score |= UNASSIGNED;
+#ifdef _TARGET_ARM_
+            if (currentInterval->registerType == TYP_DOUBLE)
+            {
+                RegRecord* anotherPhysRegRecord = nullptr;
+                anotherPhysRegRecord            = findAnotherHalfRegRec(physRegRecord);
+                if (anotherPhysRegRecord->assignedInterval == nullptr ||
+                    anotherPhysRegRecord->assignedInterval->getNextRefLocation() > lastLocation)
+                {
+                    // In case the currentInterval has a double register,
+                    // the score would be set UNASSIGNED if another register is also not set.
+                    score |= UNASSIGNED;
+                }
+            }
+            else
+#endif
+                score |= UNASSIGNED;
```

Comment:
This should be abstracted with a method, e.g. `isUnassigned(Interval* currentInterval, `RegisterType` regType)`.